staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go

```diff
@@ -180,7 +183,10 @@ func (o *CreateDeploymentOptions) Validate() error {
 
 // Run performs the execution of 'create deployment' sub command
 func (o *CreateDeploymentOptions) Run() error {
-	deploy := o.createDeployment()
+	deploy, err := o.createDeployment()
```

Comment:
Same comment from above, instead of changing the "createDeployment" IMO the best approach here is to create it, and then merge the maps :)